library/L2_HAL/sensors/environment/light/temt6000x01.hpp

```diff
@@ -7,27 +7,34 @@
 
 namespace sjsu
 {
+/// The TEMT6000X01 is an ambient light sensor that provides light readings
+/// that can be read using the ADC peripheral.
 class Temt6000x01 final : public LightSensor
 {
  public:
-  explicit constexpr Temt6000x01(Adc & adc,
+  /// @param adc The ADC peripheral used to read the sensor data.
+  /// @param adc_reference_voltage The reference voltage of the ADC. This value
+  ///                              will vary based on the chosen MCU.
+  /// @param pull_down_resistance The resistance of the pull down resistor used
+  ///                             in the circuit.
+  explicit constexpr Temt6000x01(sjsu::Adc & adc,
                                  units::voltage::volt_t adc_reference_voltage,
                                  units::impedance::ohm_t pull_down_resistance)
       : adc_(adc),
         kAdcReferenceVoltage(adc_reference_voltage),
         kPullDownResistance(pull_down_resistance),
         adc_resolution_(0)
   {
-    adc_resolution_ = powf(2, adc_.GetActiveBits()) - 1;
+    adc_resolution_ = static_cast<float>((1 << adc_.GetActiveBits()) - 1);
```

Comment:
Getting the maximum of the adc should be a built in utility of the interface. So I'm adding an issue to resolve this as well as improve the API for ADC #1011. No work required on your side though.